tests/testthat/test-impute.r

```diff
@@ -4,21 +4,38 @@
 # 2. Only a subset of a dataframe
 # 3. Factor inputs
 # 4. Factor inputs with 2 modes (uses alphabetical ranking)
+# 5. Irrelevant list of impute values throws warning but doesn't get in the way
+# 6. When a non-default value is provided it is used and doesn't interfere with defaults
+  # a. All columns are provided in unnamed list
+  # b. Named list of imputeVals in different order than df columns is provided
+  # c. Only the non-default column is provided
+# 7. Impute values with names not in df are ignored with warning
 # Test imputation using 2 function calls (first saves the values, second applies them):
 # 1. Entire dataframe, character and numeric inputs
 # 2. Only a subset of a dataframe
 # 3. Factor inputs
 # 4. Factor inputs with 2 modes (uses alphabetical ranking)
 # 5. Calculate values for 4 columns, apply them correctly to 2 columns.
-# 6. When an impute value would be a new level to be applied.
+# 6. When an impute value would be a new level to be applied and another is irrelevant
 # Error Handling:
 # 1. Different length args
 # 2. An all-NA column
 # 3. No effect on full columns of factors, numbers, and characters
 # 4. When inputs are not correct types
+# 5. Unnamed list not of same length as df produces error
+# 6. Partially named list produces error
 
 context("Checking that columns and dataframes are imputed correctly")
 
+# One source and two target data frames to test on
+original <- data.frame(x = c(3, -3, NA, 0), y = c("A", "A", "B", NA))
+default_target <- list(df = data.frame(x = c(3, -3, 0, 0),
+                                       y = c("A", "A", "B", "A")),
+                       imputeVals = list(x = 0, y = "A"))
+custom_target <- list(df = data.frame(x = c(3, -3, 0, 0),
+                                      y = c("A", "A", "B", "C")),
+                      imputeVals = list(x = 0, y = "C"))
+
 test_that("entire dataframe is imputed correctly", {
```

Comment:
I see what you mean, and that is a big problem. On the other hand, creating new objects in each test is such a violation of the don't-repeat-yourself maxim and makes it so much harder to make changes, I feel like there must be a better way. This is a common R gotchya, expecting a local variable but getting a global instead, so I'm guessing smart folks have thought hard about this. I'll make these changes for now, but I'm adding this to the "discuss eventually" list. :/